packages/core/src/Stores/client-store.js

```diff
@@ -1088,20 +1083,33 @@ export default class ClientStore extends BaseStore {
         // login_new_user is populated only on virtual sign-up
         let obj_params = {};
         const search = window.location.search;
+
         if (search) {
+            let search_params = new URLSearchParams(window.location.search);
             const arr_params = window.location.search.substr(1).split('&');
+
             arr_params.forEach(function(param) {
                 if (param) {
                     const param_value = param.split('=');
-                    if (param_value) {
-                        obj_params[param_value[0]] = param_value[1];
+                    const [key, value] = param_value;
+
+                    const account_keys = ['acct', 'token', 'cur'];
+                    const is_account_param = account_keys.some(account_key => key?.includes(account_key));
+
+                    if (is_account_param) {
+                        search_params.delete(key);
+                        obj_params[key] = value;
                     }
                 }
             });
+
+            // replace account query params - but keep other query params (e.g. utm)
+            search_params = search_params?.toString();
+            const search_param_without_account = search_params ? '?' + search_params : '/';
+            history.replaceState(null, null, search_param_without_account);
```

Comment:
Since we're using the URLSearchParams API now, can we make use of its provided interfaces to rewrite this?

Reply:
@mahdipourziaei-binary sure, rewrote it.